tests/test_solver_manager.py

```diff
@@ -126,11 +127,12 @@ def get_problem(problem_id):
                       .with_problem_finder(get_problem)).run()
         assert_solver_run(solver_manager, solver_job)
 
-        lock.acquire()
-        solver_job = (solver_manager.solve_builder()
-                      .with_problem_id(1)
-                      .with_problem_finder(get_problem)).run()
-        assert_problem_change_solver_run(solver_manager, solver_job)
+        # Disabled ; Flaky
```

Comment:
The PR description mentions that the issue occurs if the problem change happens before the solving process has started. What if we add a new flag `waitForSolver` (false by default) to `assert_problem_change_solver_run`? This could ensure the status is not `SolverStatus.NOT_SOLVING` before validating the first assertion.

Reply:
We use locks; I tried a bunch of things to reduce flakiness, but none work. The flakiness also exists in main, but was less likely to occur since the tests were only run once.